contrib-compiler.md

```diff
@@ -0,0 +1,99 @@
+
+
+Thanks for interest in contributions.
+
+Note currently our philosphy is to make release build as easy as possible, while dev build is fairly complex at this time.
```

Comment:
I think it's really cool that you share the reasoning behind the tradeoffs (making releases fast and easy) and the internals of how the build works